hopsworks-common/pom.xml

```diff
@@ -109,7 +109,7 @@
     <dependency>
       <groupId>org.apache.kafka</groupId>
       <artifactId>kafka-clients</artifactId>
-      <version>1.1.0</version>
+      <version>2.3.0</version>
```

Comment:
@kai-chi can you check if this new version brings new dependencies, such as new jackson version, in Hopsworks?

Reply:
yes, it takes jackson from 2.9.4 to 2.9.9 but no more

Reply:
@kai-chi Can you do a mvn dependency tree check to see if Hopsworks has any other jackson dependencies? We need to minimize the risk of having conflicting versions. 
Also, do we use any code `kafka-clients` methods in Hopsworks that would make use of jackson? Probably not, so if that's the case we could potentially exclude jackson.

Reply:
I added exclusions to kafka dependency. the mvn dependency tree shows that:
- kafka-clients introduces new dependency com.github.luben:zstd-jni:jar:1.4.0-1 and takes org.lz4:lz4-java from 1.4 to 1.6.0
- kafka takes jackson-databind from 2.9.4 to 2.9.9 and doesn't introduce anything

Reply:
Great, then we can merge this after we test that Hopsworks code using kafka and kafka-clients is not affected by that

Reply:
tests passed